### PR TITLE
Admin Guide updates for SES7

### LINKED ISSUE #306: [doc] 6.4.5.1  Specification
6.4.5.1  Specification

https://susedoc.github.io/doc-ses/master/html/ses-deployment/deploy-cephadm.html#drive-groups-specs

>  For FileStore setups, drive_groups.yml can be as follows: 

No filestore in SES7, this section can be dropped. We need to info user to migrate filestore OSDs to bluestore in upgrade section.

### LINKED ISSUE #370: Suspicious Admin Guide sections (mainly from DeepSea perspective)
This issue lists sections which are obviously wrong and need to be either removed from the guide or updated to match SES7:

- [x] --- Admin Tasks ---
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#ds-osd-recover
- manually deploy OSDs: ceph-volume to create devices, deploy OSDs from containers, we need a seprate JIRA issue here https://jira.suse.com/browse/SES-2110
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#moving-saltmaster
- deploy new node + restore salt-master + ceph-salt import  from exported backup
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#deepsea-rolling-updates
- underlying OS: ceph-salt update (https://github.com/ceph/ceph-salt/pull/303)
- containerized services: cephadm (https://docs.ceph.com/docs/master/cephadm/upgrade/)
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#sec-salt-cluster-reboot
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#admin-apparmor
- remove the whole section
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#deepsea-ceph-purge
- ceph-salt purge

- [x] --- Cluster Backup ---
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#sec-deployment-backup-deepsea
- remove
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#backup-config-files
- no openattic, no ceph.conf
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#restore-ceph
- remove step 4, 9-end

- [ ] --- Monitoring & Alerting ---
- [ ] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#monitoring-alerting
- talk to Patrick Seidensal @p-se 
- not as flexible as in SES7, config options must be passed to cephadm command, it re-generates their config files
- https://jira.suse.com/browse/SES-1194
- move '7.5 Troubleshooting Alerts' to Troubleshooting guide
- replace systemd units with `ceph orch demon restart osd.123`

- [x] --- RGW ---
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#ogw-deploy
- YAML based import
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#ceph-rgw-operating
- dont use systemd unit files
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#ogw-selfcert (is there a common SSL section?)
- remove DS specific stuff + https://confluence.suse.com/pages/viewpage.action?pageId=505119148 has most of the info
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#ogw-ssl-simple, https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#ogw-ssl-advanced
- not needed, ceph-salt takes care
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#ceph-rgw-sync-zones-configure (bullet 5)
- dont use systemd unit files but `ceph orch restart rgw.realm.zone`
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#ceph-rgw-fed-masterzone-startrgw
- dont use systemd unit files but `ceph orch restart rgw.realm.zone`
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#ceph-rgw-fed-secondzone-startrgw
- dont use systemd unit files but `ceph orch restart rgw.realm.zone`
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#ceph-rgw-failover-dr (bullets 3,4,7)
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#ceph-rgw-single-site-multi (bullet 7)
- move to the upgrade process
- [x] 12.13.3.3 Update the Ceph Configuration File
- remove the section
- [x] 6.5.3.4 Deploy Object Gateways
- steps 1-3 are optional, if realm, zone do not exist, they are created 

12.14 Load Balancing the Object Gateway Servers with HAProxy is going to be handled by orchestrator automatically

- [ ] --- NFS Ganesha ---
- [ ] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#ceph-nfsganesha-config
- Bug filed to ensure we have man page access #SES-2108
- [ ] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#ceph-nfsganesha-config-service-general
- @mgfritch  required here. The files are wrong, and the config files are actually stored in the middle layer. See: https://confluence.suse.com/display/SUSEEnterpriseStorage/nfs-ganesha
- [ ] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#ceph-nfsganesha-customrole-rgw-multiusers
- @mgfritch  to review. Config files are all different now. https://github.com/ceph/ceph/blob/master/src/pybind/mgr/cephadm/templates/services/nfs/ganesha.conf.j2
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#ceph-nfsganesha-customrole-rgw-cephfs
- Remove chapter completely.
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#ceph-nfsganesha-services
- Use the `ceph orch restart nfs.<service.id>` command instead

- [x] --- Ceph Point Releases ---
- [x] https://susedoc.github.io/doc-ses/master/single-html/ses-admin/#id-1.4.9

### PR DESCRIPTION
**Pull request by tbazant**
This is a major update to the Admin guide.
It aims to remove DeepSea related stuff and align the content with cephadm.
It spans across several topics, therefore please pick yours and review it :-)

closes #306 
closes #370

### CODE REVIEW

File: xml/admin_ceph_gateway.xml [hunk 1]
```diff
@@ -93,87 +93,30 @@
...
+   identifying the service name with the <command>ceph orch ps</command>
+   command, and running the following command for operating services, for
+   example:
   </para>
 
+<screen>ceph orch daemon restart <replaceable>OGW_SERVICE_NAME</replaceable></screen>
+
   <para>
-   The following subcommands are supported for the &ogw; service:
+   Refer to <xref linkend="cha-ceph-operating"/> for complete information about
```
Comment (lproven) [inline xml/admin_ceph_gateway.xml:117]:
```suggestion
   Refer to <xref linkend="cha-ceph-operating"/> for more information about
```

Comment (lproven) [reply xml/admin_ceph_gateway.xml:117]: "Complete" seems a bit optimistic. :-)

File: xml/admin_ceph_gateway.xml [hunk 2]
```diff
@@ -594,20 +532,16 @@ sudo zypper refresh</screen>
...
   <title>Enabling HTTPS/SSL for &ogw;s</title>
 
   <para>
-   To enable the default &ogw; role to communicate securely using SSL, you need
-   to either have a CA issued certificate or create a self-signed one. There
-   are two ways to configure &ogw; with HTTPS enabled&mdash;a simple way that
-   makes use of the default settings, and an advanced way that lets you fine
-   tune HTTPS related settings.
+   To enable the &ogw; to communicate securely using SSL, you need to either
+   have a CA issued certificate or create a self-signed one.
```
Comment (lproven) [inline xml/admin_ceph_gateway.xml:544]:
```suggestion
   have a CA-issued certificate or create a self-signed one.
```

File: xml/admin_ceph_gateway.xml [hunk 3]
```diff
@@ -617,13 +551,6 @@ sudo zypper refresh</screen>
      Skip this section if you already have a valid certificate signed by CA.
```
Comment (lproven) [inline xml/admin_ceph_gateway.xml:551]:
```suggestion
     Skip this section if you already have a valid certificate signed by a CA.
```

File: xml/admin_ds_backup.xml [hunk 4]
```diff
@@ -24,186 +24,147 @@
...
-  </para>
-
-  <note>
-   <title>Security of the Salt Master Private Key</title>
+  <sect2 xml:id="backrest-ceph-cephsalt">
+   <title>Back Up &cephsalt; Configuration</title>
    <para>
-    Make sure that the backup of the &smaster; private key is stored in a safe
-    location. The &smaster; key can be used to manipulate all cluster nodes.
+    Export cluster configuration. Find more information in
```
Comment (lproven) [inline xml/admin_ds_backup.xml:31]: Export _the_ cluster configuration?

File: xml/admin_ds_backup.xml [hunk 5]
```diff
@@ -24,186 +24,147 @@
...
-
- </sect2>
- <sect2 xml:id="backup-config-files">
-  <title>Back Up Custom Configurations</title>
+  <sect2 xml:id="backup-ceph">
+   <title>Back Up &ceph; Configuration</title>
+   <para>
+    Back up the <filename>/etc/ceph</filename> directory. It contains crucial
+    cluster configuration. You will need the backup of
+    <filename>/etc/ceph</filename> for example when you need to replace the
```
Comment (lproven) [inline xml/admin_ds_backup.xml:41]:
```suggestion
    Back up the <filename>/etc/ceph</filename> directory. It contains crucial
    cluster configuration. For example, you will need a backup of
    <filename>/etc/ceph</filename> when you need to replace the
```

File: xml/admin_ds_backup.xml [hunk 6]
```diff
@@ -24,186 +24,147 @@
...
-    </para>
-   </listitem>
-  </itemizedlist>
- </sect2>
+    </listitem>
+    <listitem>
+     <para>
+      &sgw; configuration. If you are using a single gateway, backup
+      <filename>/etc/samba/smb.conf</filename>. If you are using HA setup,
+      backup also CTDB and Pacemaker configuration files. Refer to
```
Comment (lproven) [inline xml/admin_ds_backup.xml:102]:
```suggestion
      <filename>/etc/samba/smb.conf</filename>. If you are using a HA setup,
      also backup the CTDB and Pacemaker configuration files. Refer to
```

File: xml/admin_ds_backup.xml [hunk 7]
```diff
@@ -24,186 +24,147 @@
...
+      <xref linkend="cha-ses-cifs"/> for details on what configuration is used
+      by &sgw;s.
+     </para>
+    </listitem>
+    <listitem>
+     <para>
+      &ganesha; configuration. Only needed when using HA setup. Refer to
```
Comment (lproven) [inline xml/admin_ds_backup.xml:109]:
```suggestion
      &ganesha; configuration. Only needed when using a HA setup. Refer to
```

File: xml/admin_intro.xml [hunk 8]
```diff
@@ -28,15 +28,15 @@
   commodity hardware platforms.
  </para>
  <para>
-  This guide helps you understand the concept of the &productname;
-  &productnumber; with the main focus on managing and administrating the &ceph;
-  infrastructure. It also demonstrates how to use &ceph; with other related
-  solutions, such as &ostack; or &kvm;.
+  This guide focuses on routine tasks that you as an administrator need to take
+  care of after the basic &ceph; cluster has been deployed (day 2 operations).
+  It also describes all supported ways to access data stored in a &ceph;
```
Comment (lproven) [inline xml/admin_intro.xml:33]:
```suggestion
  It also describes all the supported ways to access data stored in a &ceph;
```

File: xml/admin_intro.xml [hunk 9]
```diff
@@ -28,15 +28,15 @@
...
+  cluster.
  </para>
  <para>
-  Many chapters in this manual contain links to additional documentation
-  resources. These include additional documentation that is available on the
-  system as well as documentation available on the Internet.
+  Chapters in this manual contain links to additional documentation resources.
```
Comment (lproven) [inline xml/admin_intro.xml:37]:
```suggestion
  The chapters in this manual contain links to additional documentation resources.
```

File: xml/admin_intro.xml [hunk 10]
```diff
@@ -28,15 +28,15 @@
...
+  These include additional documentation that is available on the system as
```
Comment (lproven) [inline xml/admin_intro.xml:38]:
```suggestion
  These include additional documentation that is available on the system, as
```

File: xml/admin_saltcluster.xml [hunk 11]
```diff
@@ -441,77 +441,91 @@ osd.1 node3 3 2020-04-22 19:28:34.201685
...
+<!-- https://docs.ceph.com/docs/master/cephadm/upgrade/ -->
 
-  <sect2>
-   <title>Downtime of &ceph; Services</title>
+  <title>Upgrading &ceph;</title>
+
+  <para>
+   You can instruct &cephadm; to upgrade &ceph; from one bugfix release to
+   another. The automated upgrade of &ceph; services respects the recommended
+   order&mdash;it starts with &mgr;s, &mon;s, and then continues on other
```
Comment (lproven) [inline xml/admin_saltcluster.xml:471]:
```suggestion
   order&mdash;it starts with &mgr;s, &mon;s, and then continues on to other
```

File: xml/common_intro_feedback_i.xml [hunk 12]
```diff
@@ -10,69 +10,43 @@
...
      <link xlink:href="http://www.suse.com/support/"/>.
     </para>
     <para>
-     To report bugs for a product component, log in to the &ncc; from
-     <link xlink:href="http://www.suse.com/support/"/> and select <menuchoice>
-     <guimenu>My Support</guimenu> <guimenu>Service Request</guimenu>
-     </menuchoice>.
+     To open a service request, you need a subscription at SUSE Customer
+     Center. Go to <link xlink:href="https://scc.suse.com/support/requests"/>,
+     log in, and click Create New.
```
Comment (lproven) [inline xml/common_intro_feedback_i.xml:40]: Shouldn't there be some kind of markup or at least quotation marks around "Create New"?

File: xml/leftovers.xml [hunk 13]
```diff
@@ -87,3 +87,74 @@ skip=$((`blockdev --getsz /dev/sdX` - 33)) | hexdump -C
 <!-- MARKING MINIONS -->
 <screen>ceph-salt config /Ceph_Cluster/Minions add doc-ses-*suse.cz</screen>
 <!-- / MARKING MINIONS -->
+
+<!-- 2020-08-11 move vvvvvvvvvvvvvvvvvvvvvv to the upgrade chapter -->
+  <sect2 xml:id="ceph-rgw-single-site-multi">
+   <title>Migrating a Single Site System to Multi-Site</title>
+   <para>
+    To migrate from a single site system with a default &zgroup; and zone to a
```
Comment (lproven) [inline xml/leftovers.xml:95]:
```suggestion
    To migrate from a single-site system with a default &zgroup; and zone to a
```

File: xml/leftovers.xml [hunk 14]
```diff
@@ -87,3 +87,74 @@ skip=$((`blockdev --getsz /dev/sdX` - 33)) | hexdump -C
...
+    <step>
+     <para>
+      Restart the &cogw;.
+     </para>
+<screen>&prompt.cephuser.ogw;systemctl restart ceph-radosgw@rgw.`hostname -s`</screen>
+    </step>
+   </procedure>
+   <para>
+    After completing this procedure, configure a secondary zone to create a
+    secondary zone in the master &zgroup;.
```
Comment (lproven) [inline xml/leftovers.xml:157]: This sentence seems redundant. Configure a secondary zone to create a secondary zone?

File: xml/admin_ceph_gateway.xml [hunk 15]
```diff
@@ -3152,10 +2986,11 @@ rgw_zone=us-east</screen>
     <important>
      <para>
       The following steps assume a multi-site configuration using
-      newly-installed systems that are not storing data. <emphasis
-       role="bold">Do not delete</emphasis> the default zone and its pools if
-      you are already using it to store data, or the data will be lost and
-      unrecoverable.
+      newly-installed systems that are not storing data.
```
Comment (tbazant) [inline xml/admin_ceph_gateway.xml:2989]:
```suggestion
      newly-installed systems that are not yet storing data.
```

File: xml/admin_ds_backup.xml (hunk 4 above)
Comment (tbazant) [inline xml/admin_ds_backup.xml:31]:
```suggestion
    Export the cluster configuration. Find more information in
```

File: xml/admin_ds_backup.xml (hunk 7 above)
Comment (tbazant) [inline xml/admin_ds_backup.xml:109]:
```suggestion
      &ganesha; configuration. Only needed when using an HA setup. Refer to
```

File: xml/common_intro_feedback_i.xml (hunk 12 above)
Comment (tbazant) [inline xml/common_intro_feedback_i.xml:40]:
```suggestion
     log in, and click <guimenu>Create New</guimenu>.
```

File: xml/leftovers.xml [hunk 16]
```diff
@@ -87,3 +87,74 @@ skip=$((`blockdev --getsz /dev/sdX` - 33)) | hexdump -C
...
+    </step>
+    <step>
+     <para>
+      Restart the &cogw;.
+     </para>
+<screen>&prompt.cephuser.ogw;systemctl restart ceph-radosgw@rgw.`hostname -s`</screen>
+    </step>
+   </procedure>
+   <para>
+    After completing this procedure, configure a secondary zone to create a
```
Comment (tbazant) [inline xml/leftovers.xml:156]:
```suggestion
    After completing this procedure, configure a 
```

**Comment by sebastian-philipp** [comment]
do you have a render of the docs? I have a hard time parsing the xml diff in my head

**Comment by tbazant** [comment]
> do you have a render of the docs? I have a hard time parsing the xml diff in my head

@sebastian-philipp  the Admin Guide PDF is here - 
[book-storage-admin_color_en.pdf](https://github.com/SUSE/doc-ses/files/5090232/book-storage-admin_color_en.pdf)

File: xml/admin_saltcluster.xml [hunk 17]
```diff
@@ -205,6 +205,764 @@ ses-min2   /dev/vdc  hdd  8192M  450575 True
...
+     you need to avoid this behavior, add the <option>--unmanaged=true</option>
+     option when creating OSDs:
+    </para>
+<screen>&prompt.cephuser;ceph orch apply osd -i <filename>drive_groups.yml</filename> --unmanaged=true</screen>
+    <para>
+     Alternatively, if you have already created OSDs without the
+     <option>--unmanaged=true</option>, you can disable the automatic OSD
+     creation using the following command:
+    </para>
+<screen>&prompt.cephuser;ceph orch osd spec --service-name  osd.all-available-devices --unmanaged</screen>
```
Comment (jschmid1) [inline xml/admin_saltcluster.xml:250]: The argument  to the `---service-name` parameter is not static. It needs to be in format of `service_type.service_id`

File: xml/admin_saltcluster.xml [hunk 18]
```diff
@@ -205,6 +205,764 @@ ses-min2   /dev/vdc  hdd  8192M  450575 True
...
+  host_pattern: '*'
+data_devices:
+  drive_spec: <replaceable>DEVICE_SPECIFICATION</replaceable>
+db_devices:
+  drive_spec: <replaceable>DEVICE_SPECIFICATION</replaceable>
+wal_devices:
+  drive_spec: <replaceable>DEVICE_SPECIFICATION</replaceable>
+block_wal_size: '5G'  # (optional, unit suffixes permitted)
+block_db_size: '5G'   # (optional, unit suffixes permitted)
+osds_per_device: 1   # number of osd daemons per device
```
Comment (jschmid1) [inline xml/admin_saltcluster.xml:270]: Let's leave that out for now, as it's not fully supported I think

File: xml/admin_saltcluster.xml (hunk 17 above)
Comment (tbazant) [reply xml/admin_saltcluster.xml:250]: does it apply to services other than `osd` as well?

File: xml/admin_saltcluster.xml [hunk 19]
```diff
@@ -160,171 +160,897 @@ placement:
...
+     you need to avoid this behavior, add the <option>--unmanaged=true</option>
+     option when creating OSDs:
+    </para>
+<screen>&prompt.cephuser;ceph orch apply osd -i <filename>drive_groups.yml</filename> --unmanaged=true</screen>
+    <para>
+     Alternatively, if you have already created OSDs without the
+     <option>--unmanaged=true</option>, you can disable the automatic OSD
+     creation using the following command:
+    </para>
+<screen>&prompt.cephuser;ceph orch osd spec --service-name  osd.all-available-devices --unmanaged</screen>
```
Comment (tbazant) [inline xml/admin_saltcluster.xml:250]:
```suggestion
<screen>&prompt.cephuser;ceph orch osd spec --service-name  osd.<replaceable>OSD_SERVICE_ID</replaceable> --unmanaged</screen>
```

File: xml/admin_saltcluster.xml (hunk 18 above)
Comment (tbazant) [reply xml/admin_saltcluster.xml:270]: > Let's leave that out for now, as it's not fully supported I think

what exactly? `osds_per_device` or `block_db_size` and `block_wal_size` as well?

File: xml/admin_saltcluster.xml (hunk 17 above)
Comment (tbazant) [reply xml/admin_saltcluster.xml:250]: > The argument to the `---service-name` parameter is not static. It needs to be in format of `service_type.service_id`

Does it apply to services other than osd as well?

File: xml/admin_saltcluster.xml [hunk 20]
```diff
@@ -160,186 +160,916 @@ placement:
...
+  host_pattern: '*'
+data_devices:
+  drive_spec: <replaceable>DEVICE_SPECIFICATION</replaceable>
+db_devices:
+  drive_spec: <replaceable>DEVICE_SPECIFICATION</replaceable>
+wal_devices:
+  drive_spec: <replaceable>DEVICE_SPECIFICATION</replaceable>
+block_wal_size: '5G'  # (optional, unit suffixes permitted)
+block_db_size: '5G'   # (optional, unit suffixes permitted)
+osds_per_device: 1   # number of osd daemons per device
```
Comment (tbazant) [inline xml/admin_saltcluster.xml:270]:
```suggestion
<!-- osds_per_device: 1   # number of osd daemons per device -->
```

**Comment by lproven** [comment]
I tried to build this to give it another read-through, but I get many screensful of this error:
> /doc-ses/build/.profiled/sles/windows-ceph.xml:11:34: error: element "section" not allowed anywhere; expected the element end-tag or element "annotation", "bridgehead", "calloutlist", "example", "figure", "formalpara", "glossary", "glosslist", "important", "index", "indexterm", "informalexample", "informalfigure", "informaltable", "itemizedlist", "note", "orderedlist", "para", "procedure", "programlisting", "qandaset", "refentry", "refsect1", "remark", "screen", "sect1", "simplelist", "table", "task", "tip", "topic", "variablelist", "warning" or "xi:include"

**Comment by tbazant** [comment]
> I tried to build this to give it another read-through, but I get many screensful of this error:
> 
> > /doc-ses/build/.profiled/sles/windows-ceph.xml:11:34: error: element "section" not allowed anywhere; expected the element end-tag or element "annotation", "bridgehead", "calloutlist", "example", "figure", "formalpara", "glossary", "glosslist", "important", "index", "indexterm", "informalexample", "informalfigure", "informaltable", "itemizedlist", "note", "orderedlist", "para", "procedure", "programlisting", "qandaset", "refentry", "refsect1", "remark", "screen", "sect1", "simplelist", "table", "task", "tip", "topic", "variablelist", "warning" or "xi:include"

@lproven if you build it now inside the `tbazant-admin_guide_updates_ses7` branch, it validates ok

**Pull request merged**